pkg/proxy/iptables/proxier.go

```diff
@@ -575,48 +572,31 @@ func (proxier *Proxier) OnServiceDelete(service *v1.Service) {
 func (proxier *Proxier) OnServiceSynced() {
 	proxier.mu.Lock()
 	proxier.servicesSynced = true
-	if utilfeature.DefaultFeatureGate.Enabled(features.EndpointSliceProxying) {
-		proxier.setInitialized(proxier.endpointSlicesSynced)
-	} else {
-		proxier.setInitialized(proxier.endpointsSynced)
-	}
+	proxier.setInitialized(proxier.endpointSlicesSynced)
 	proxier.mu.Unlock()
 
 	// Sync unconditionally - this is called once per lifetime.
 	proxier.syncProxyRules()
 }
 
+// iptables proxier only uses EndpointSlice, the following methods
+// exist to implement the Proxier interface but are noops
```

Comment:
Should we remove them from the interface?

Reply:
Since `userspace` still exists, we can't remove it from the interface just yet. Once we remove support for `userspace` we remove from the interface as well.

Reply:
cc @danwinship

Reply:
Definitely no need to block this PR, but both `userspace` and `win_userspace` proxiers actually appear to have old copies of this code contained locally so it is likely safe to remove these.

https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/userspace/proxier.go#L668
https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/winuserspace/proxier.go#L438

Reply:
It is possible, but the change would require changes to the expected interfaces in the ProxyServers in `cmd/kube-proxy` and changing the logic to keep track of the userspace proxiers vs the other proxiers. I feel like instead of adding that, it may be better to remove that path completely if userspace is no longer an option we want to support.